### PR TITLE
Add Erigon to gnosis tests

### PR DESCRIPTION
**Pull request by stdevMac**
- Added Erigon to Gnosis Hive tests

### CODE REVIEW

File: .github/workflows/dashboard_erigon.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+name: Daily Run Erigon 
+
+on:
+  schedule:
+    - cron: '0 0 * * *'  # Run every day at midnight UTC
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - master
+    types: [ closed ]
```
Comment (4rgon4ut) [inline .github/workflows/dashboard_erigon.yml:10]: optional: Maybe we should add run on push to master to see all tests passed before the merge?

File: simulators/ethereum/engine/helper/core.go [hunk 2]
```diff
@@ -335,3 +335,177 @@ func (n *NethermindChainSpec) ToBlock() *types.Block {
...
+
+func (v *ErigonGenesis) AllocGenesis(address common.Address, account Account) {
+	v.ErigonAlloc[address.Hex()] = ErigonAccount{
+		Balance:     account.Balance().String(),
+		Constructor: account.Constructor(),
+	}
+}
+
+func (v *ErigonGenesis) UpdateTimestamp(timestamp string) {
+	//v.ErigonConfig.ShanghaiTimestamp =
```
Comment (4rgon4ut) [inline simulators/ethereum/engine/helper/core.go:485]:
```suggestion
	panic("implement me")
```

File: simulators/ethereum/engine/test/spec.go [hunk 3]
```diff
@@ -108,15 +108,27 @@ func (s Spec) GetForkConfig() ForkConfig {
...
 }
 
+func GenesisFactory(clientName string) helper.Genesis {
+	switch clientName {
+	case "erigon_":
+		return &helper.ErigonGenesis{}
+	case "nethermind_":
+		return &helper.NethermindChainSpec{}
+	default:
+		return nil
```
Comment (4rgon4ut) [inline simulators/ethereum/engine/test/spec.go:118]: nit: I guess it make sense to panic or return error here as upcoming `genesis.UnmarshalJSON` will cause nil pointer exception anyways, does it ?

**Review by 4rgon4ut — APPROVED** [review]
LGTM. Great job @stdevMac! Dropped down couple optionals.
Does uncommented reorg tests work with erigon too?

File: simulators/ethereum/engine/test/spec.go (hunk 3 above)
Comment (stdevMac) [reply simulators/ethereum/engine/test/spec.go:118]: It will break in the next step, that's ok for now

Comment (4rgon4ut) [reply simulators/ethereum/engine/test/spec.go:118]: Yes and I meant that we can panic here to be sure where is the problem exactly (i.e. `unsupported client provided`)

Comment (stdevMac) [reply simulators/ethereum/engine/test/spec.go]: Done

**Pull request merged**